Userland/Libraries/LibJS/Runtime/VM.cpp

```diff
@@ -207,18 +207,15 @@ void VM::assign(const NonnullRefPtr<BindingPattern>& target, Value value, Global
         if (!iterator)
             return;
 
-        size_t index = 0;
-        while (true) {
+        for (size_t i = 0; i < binding.properties.size(); i++) {
             if (exception())
                 return;
 
-            if (index >= binding.properties.size())
-                break;
-
-            auto pattern_property = binding.properties[index];
-            ++index;
+            auto pattern_property = binding.properties[i];
```

Comment:
Let's not copy this, it could involve three different refcount increment/decrements